assets/differentialabundance_report.Rmd

```diff
@@ -815,17 +815,23 @@ if (any(unlist(params[paste0(possible_gene_set_methods, '_run')]))){
     if (unlist(params[paste0(gene_set_method, '_run')])){
       cat("\n### ", toupper(gene_set_method) ," {.tabset}\n")
       
-      reference_gsea_tables <- paste0(contrasts$id, '.gsea_report_for_', contrasts$reference, '.tsv')
-      target_gsea_tables <- paste0(contrasts$id, '.gsea_report_for_', contrasts$target, '.tsv')
-
-      for (i in 1:nrow(contrasts)){
-        cat("\n#### ", contrast_descriptions[i], "\n")
+      for (gmt_file in strsplit(params$gsea_gene_sets, " ")[[1]]) {
+        gmt_name <- strsplit(basename(gmt_file), '.', fixed=T)[[1]]
+        gmt_name <- paste0(head(gmt_name, -1), collapse=".")
         
-        target_gsea_results <- read_metadata(target_gsea_tables[i])[,c(-2,-3)]
-        print( htmltools::tagList(datatable(target_gsea_results, caption = paste0("\nTarget (", contrasts$target[i], ")\n"), rownames = FALSE) ))
+        cat("\n#### ", gmt_name ," {.tabset}\n")
```

Comment:
I'm not 100% sure about the hierarchical tabsets, I find the display a bit confusing, but let's go with it for now.

Reply:
I know what you mean but dunno what would be a better alternative